File: src/HomeFinder.Entity/DB/Item.cs

namespace HomeFinder.Entity.DB
{
    public class Item
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        public string JANCode { get; set; } = string.Empty;

        public decimal Price { get; set; }

        public Picture? Picture { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public int CategoryId { get; set; }
        public Category? Category { get; set; }
    }
}

File: src/HomeFinder.UI/src/models/item.ts
export type Item = {
  id: string;
  name: string;
  quantity: number;
  manufacturer?: string | null;
  description?: string | null;
  note?: string | null;
  barcode?: string | null;
  price?: number | null;
  categoryId?: string | null;
  categoryName?: string | null;
  roomId?: string | null;
  roomDisplayName?: string | null;
  shelfId?: string | null;
  shelfDisplayName?: string | null;
  createdAt: string;
  updatedAt: string;
  imageId?: string | null;
  imageUrl?: string | null;
};

export type PagedItemsResponse = {
  items: Item[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
};
